lib/meadow/pipeline/actions/transcode_complete.ex

```diff
@@ -0,0 +1,77 @@
+defmodule Meadow.Pipeline.Actions.TranscodeComplete do
+  @moduledoc """
+  Action to handle job state change messages from MediaConvert
+  """
+
+  alias Meadow.Data.{ActionStates, FileSets}
+  alias Meadow.Pipeline.Actions.Common
+
+  use Sequins.Pipeline.Action
+  use Meadow.Utils.Logging
+
+  require Logger
+
+  @impl true
+  def process(%{detail_type: "MediaConvert Job State Change"} = message, _attributes) do
+    with {data, attributes} <- parse(message) do
+      process(data, attributes)
+    end
+  end
+
+  def process(%{file_set_id: file_set_id} = data, attributes) do
+    with_log_metadata module: __MODULE__, id: data.file_set_id do
+      with {status, response} <-
+             file_set_id |> FileSets.get_file_set() |> process_mediaconvert_response(data) do
+        Common.update_progress(data, attributes, {status, response})
+        {status, response, attributes}
+      end
+    end
+  end
+
+  defp parse(message) do
+    {%{
+       file_set_id: message |> get_in([:detail, :user_metadata, :file_set_id]),
+       status: message |> get_in([:detail, :status]),
+       error: message |> get_in([:detail, :error_message]),
+       playlist: message |> extract_playlist()
+     },
+     message
+     |> get_in([:detail, :user_metadata])
+     |> Map.delete(:file_set_id)}
+  end
+
+  defp extract_playlist(message) do
+    case message |> get_in([:detail, :output_group_details]) do
+      [detail | _] ->
+        detail |> get_in([:playlist_file_paths, Access.at(-1)])
+
+      _ ->
+        nil
+    end
+  end
+
+  defp process_mediaconvert_response(nil, %{file_set_id: file_set_id}) do
+    Logger.warn(
+      "Marking #{__MODULE__} for #{file_set_id} as error because the file set was not found"
+    )
+
+    {:error, "FileSet #{file_set_id} not found"}
+  end
+
+  defp process_mediaconvert_response(file_set, %{status: "skip"}) do
```

Comment:
The idea of sending a file set through all the actions and skipping the action and marking "n/a" if the file set was of the wrong mime-type, was the idea we had before we settled on the Dispatcher idea. Now the dispatcher is the one responsible for making determinations based on mime-type. 

Now that the Dispatcher is merged, I think this should come out (along with the associated code from the CreateTranscodeJob as well), unless we plan to implement the same pattern for all the actions. It doesn't seem right that this should behave differently than all the other actions and put a different status in the d/b. 

I think it should error like CreatePyramidTiff does if you pass it a text file.

Reply:
Agreed. I made a similar comment on the CreateTranscodeJob PR and forgot to look back at it when writing this one. I’ll remove it.

Reply:
I've removed both the mime type test in `CreateTranscodeJob` and the `skip` response handling from `TranscodeComplete`.